airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -91,25 +91,28 @@ def upload_file_if_changed(
 
 
 def _latest_upload(metadata: ConnectorMetadataDefinitionV0, bucket: storage.bucket.Bucket, metadata_file_path: Path) -> Tuple[bool, str]:
-    version_path = get_metadata_remote_file_path(metadata.data.dockerRepository, metadata.data.dockerImageTag)
     latest_path = get_metadata_remote_file_path(metadata.data.dockerRepository, "latest")
-    latest_icon_path = get_icon_remote_file_path(metadata.data.dockerRepository, "latest")
+    return upload_file_if_changed(metadata_file_path, bucket, latest_path, disable_cache=True)
+
+
+def _version_upload(metadata: ConnectorMetadataDefinitionV0, bucket: storage.bucket.Bucket, metadata_file_path: Path) -> Tuple[bool, str]:
+    version_path = get_metadata_remote_file_path(metadata.data.dockerRepository, metadata.data.dockerImageTag)
+    return upload_file_if_changed(metadata_file_path, bucket, version_path, disable_cache=True)
 
-    (
-        version_uploaded,
-        version_blob_id,
-    ) = upload_file_if_changed(metadata_file_path, bucket, version_path, disable_cache=True)
-    latest_uploaded, _latest_blob_id = upload_file_if_changed(metadata_file_path, bucket, latest_path, disable_cache=True)
 
-    # Replace metadata file name with icon file name
+def _icon_upload(metadata: ConnectorMetadataDefinitionV0, bucket: storage.bucket.Bucket, metadata_file_path: Path) -> Tuple[bool, str]:
     local_icon_path = metadata_file_path.parent / ICON_FILE_NAME
-    if local_icon_path.exists():
-        upload_file_if_changed(local_icon_path, bucket, latest_icon_path)
+    latest_icon_path = get_icon_remote_file_path(metadata.data.dockerRepository, "latest")
+    if not local_icon_path.exists():
+        return False, f"No Icon found at {local_icon_path}"
+    return upload_file_if_changed(local_icon_path, bucket, latest_icon_path)
 
-    return version_uploaded or latest_uploaded, version_blob_id
 
+def create_prerelease_metadata_file(metadata_file_path: Path, prerelease_tag: str) -> Path:
+    metadata, error = validate_and_load(metadata_file_path, [])
```

Comment:
👍🏻 don't validate when we load the initial file before we've overwritten it

Reply:
this will still run it through the object validator, but not perform any extra validation (like if the docker image exists). i think that's fine